components/RestaurantPage.tsx

```diff
@@ -182,122 +260,280 @@ export const RestaurantPage: React.FC<RestaurantPageProps> = ({
+    >
+      <p className="font-semibold text-xl text-dark mb-6    ">
+       {category}
+      </p>
+      
+      {filteredItems.length > 0 ? (
+      
+        <div className="   py-6 px-4 grid grid-cols-1 lg:grid-cols-2 gap-4  ">
+        {filteredItems.map(
+          (item: any) =>
+            (item.isShowing || item.enable_showing) && (
+              <div
+                key={item.name}
+                onClick={
+                  item.isAvailable
+                    ? () => handleMenuItemClick(item)
+                    : undefined
+                }
+                className={` relative flex flex-col border  border-lightdark border-solid h-32  rounded-xl bg-white px-4 py-2  ${
+                  item.isAvailable
+                    ? "hover:bg-smoke cursor-pointer duration-300"
+                    : "bg-gray-300 cursor-not-allowed"
+                }`}
+              >
+                {!item.isAvailable && (
+                  <div
+                    style={{ top: -1, right: -1 }}
+                    className=" absolute bg-primary text-white px-2 py-0 rounded-bl-lg"
+                  >
+                    OUT
+                  </div>
+                )}
+                <div className="h-1/4 flex-nowrap overflow-hidden  text-dark text-sm ">
+                  <p className=" text-dark">{item.name}</p>
+                </div>
+                <div className="  h-2/4 py-1 overflow-hidden text-dark text-sm">
+                  <p
+                    className=" overflow-hidden -webkit-line-clamp-2"
+                    style={{
+                      display: "-webkit-box",
+                      WebkitBoxOrient: "vertical",
+                      lineHeight: "1.7",
+                      padding: "0",
+                      margin: "0",
+                    }}
+                  >
+                    {item.desc}
+                  </p>
+                </div>
+                <div className="h-1/4 ">
+                  <p className=" text-dark">
+                    $
+                    {new Intl.NumberFormat("en-US", {
+                      style: "decimal",
+                      minimumFractionDigits: 2,
+                      maximumFractionDigits: 2,
+                    }).format(item.price)}
+                  </p>
+                </div>
+              </div>
+            )
+        )}
+      </div>
+      ): null}
+      
+    </div>
+    
+  );
+})}
+</div> 
+{Object.entries(restaurant.menus)
+  .filter(([category, items]) =>
+    items.some((item) =>
+      item.name.toLowerCase().includes(searchTerm.toLowerCase())
+    )
+  )
+  .filter(([category]) => category !== "properties")
+  .every(([category, items]) => {
+    const filteredItems = items.filter((item) =>
+      item.name.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+    return filteredItems.length === 0;
+  }) && <div><p className="text-gray-500 text-4xl text-center mt-10 md:mb-20   ">Item not available</p></div>}
+
```

Comment:
Might be useful in the future if logic like this is contained in a function so it can be reused and easier to read